models/pagination-2.0.0.yaml

```diff
@@ -0,0 +1,46 @@
+Pagination:
+  title: Pagination
+  type: object
+  description: |
+    'Response object to provide cursor based pagination.
+     Provides navigation links to the current, first, previous, next and last page of the resultset.
+     The value of the cursor is an opaque string that must never be inspected or constructed by clients.
+     See https://ist-group.github.io/rest-guidelines/#cursor-based-pagination
+     The total property provides the number of elements of the entire collection. The calculation of this number may have a
+     significant system and performance impact. Therefore you should avoid providing a total count unless there is a clear need to do so.'
+  properties:
+    total:
+      type: integer
+      example: 500
+      format: int32
+      minimum: 0
+      description: Optional total number of pages
+    self:
+      type: string
+      format: uri
+      example: 'https://api.ist.com/students?cursor=HUTZgriawiloJM'
+      description: Pagination link pointing to the current page.
+    first:
+      type: string
+      format: uri
+      example: 'https://api.ist.com/students?cursor=ASdagriawimke'
+      description: Pagination link pointing to the first page.
+    prev:
+      type: string
+      format: uri
+      example: 'https://api.ist.com/students?cursor=FAfsafweqsnHDDas'
+      description: Pagination link pointing to the previous page. It is not provided, if it is the first page.
+    next:
+      type: string
+      format: uri
+      example: 'https://api.ist.com/students?cursor=PFdsasqNASjd'
+      description: Pagination link pointing to the next page. It is not provided, if it is the last page.
+    last:
+      type: string
+      format: uri
+      example: 'https://api.ist.com/students?cursor=ASdasdfawiern'
+      description: Pagination link pointing to the last page.
+  required:
+    - self
+    - first
+    - last
```

Comment:
If I understood it correctly on @vershnik it can be hard to generate the link to the last page.
Maybe we shouldn't make it required but let it stay optional?

Reply:
It is not so hard, but what is real life scenario?

I load first page, and then jump to the last "to achieve what"?

Reply:
It's common to show links to last pages in UIs I would say. 
They usually look something like `< First 1 2 3 4 ... Last >` so the APIs should be able to achieve that.

Reply:
@rockymontana Yes, exactly. In the case of many pages one does not want to step page by page to reach the end. In your example there will be a loooooot of clicks to reach page 65 or 102. Even page 10 would be a number of not necessary clicks.